R/shiny.R

```diff
@@ -0,0 +1,307 @@
+#' @export
+require_oauth <- function(app, oauth_app, scopes, welcome_ui,
+  cookie_opts = cookie_options(http_only = TRUE)) {
+
+  force(oauth_app)
+  force(scopes)
+  force(welcome_ui)
+
+  httpHandler <- app$httpHandler
+  app$httpHandler <- function(req) {
+    resp <-
+      handle_oauth_callback(req, oauth_app, cookie_opts) %||%
+      handle_logged_in(req, oauth_app, httpHandler) %||%
+      handle_welcome(req, oauth_app, scopes, cookie_opts)
```

Comment:
`welcome_ui` is never used anywhere and, based purely on name, I wonder if maybe it goes here?

Reply:
Out of curiosity, why does `resp` need to be assigned?

Reply:
Yep, `welcome_ui` is not implemented yet but it will go there.

I assigned `resp` for two pretty trivial reasons: 1) to get the `handle_*` calls to be all indented together, and 2) to make it clear that there's a return value here that matters (vs. calling `handle_*` invoking some side effect, which is the way many web frameworks work).